src/controllers/user.js

```diff
@@ -34,7 +34,26 @@ const getSingleUser = async (req, res) => {
   }
 };
 
+const deleteProfile = async (req, res) => {
+  try {
+    const foundUser = await UserModel.findById(req.params.id);
+    if (foundUser) {
+      if (req.user.userId === foundUser.id) {
+        await UserModel.findByIdAndDelete(req.params.id);
```

Comment:
Clearing the cookie before returning would be a great approach.
<details> 
  <summary>The snippet we used for logout</summary>
await res.clearCookie('token');
</details>

Reply:
I wanna add something here. Since we already fetched a user from the database I don't think we need another find operation. instead if the user fulfills the authorization we can use `await foundUser.remove()`.